src/spdigger_stack.c

```diff
@@ -1940,7 +1939,38 @@ TbBool slab_is_players_land(PlayerNumber plyr_idx, MapSlabCoord slb_x, MapSlabCo
 
 long imp_already_reinforcing_at_excluding(struct Thing *creatng, MapSubtlCoord stl_x, MapSubtlCoord stl_y)
 {
-    return _DK_imp_already_reinforcing_at_excluding(creatng, stl_x, stl_y);
+    struct Map *mapblk;
+    mapblk = get_map_block_at(stl_x, stl_y);
+    struct Thing *thing;
+    long i;
+    unsigned long k;
+    k = 0;
+    i = get_mapwho_thing_index(mapblk);
+    while (i != 0)
+    {
+        thing = thing_get(i);
+        TRACE_THING(thing);
+        if (thing_is_invalid(thing))
+        {
+            ERRORLOG("Jump to invalid thing detected");
+            break;
+        }
+        i = thing->next_on_mapblk;
+        // Per thing code start
+        if(thing_is_creature(thing) && thing != creatng && (thing->alloc_flags & TAlF_IsInLimbo) == 0 && (thing->state_flags & TF1_InCtrldLimbo) == 0 && thing->active_state == CrSt_ImpReinforces)
```

Comment:
How's this for readability:
 if (thing_is_creature(thing) && (thing != creatng) && (!thing_is_picked_up(thing)) && (thing->active_state == CrSt_ImpReinforces))
 
I have to say I dislike `(thing != creatng)` because that reads as 'this thing is not a creature', where it actually is 'this new creature is not the same as the excluding creature', but I don't really have a better alternative either. Perhaps (excltng != findtng)' or something?